File: transacoes-spring/src/main/java/com/transaction/spring/entity/Transacao.java
package com.transaction.spring.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity()
@Table(name = "transacao")
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class Transacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String nome;
    private double valor;
    private String numeroConta;

    @Enumerated(EnumType.STRING)
    private TipoTransacao tipoTransacao;
}

File: transacoes-frontend/src/app/model/transacao.ts
export class Transacao {
    id!: number;
    nome!: string;
    valor!: number;
    numeroConta!: string;
    tipoTransacao!: 'CREDITO' | 'DEBITO';

    constructor(){}

}
export enum TipoTransacao {
  CREDITO = "CRÉDITO",
  DEBITO = "DÉBITO"
}
